utils.go

```diff
@@ -14,14 +43,208 @@ type Storage struct {
 
 // String implements Storager.String
 func (s *Storage) String() string {
-	panic("implement me")
+	return fmt.Sprintf(
+		"Storager minio {Name: %s, WorkDir: %s}",
+		s.bucket, s.workDir,
+	)
+}
+
+func New(pairs ...types.Pair) (types.Servicer, types.Storager, error) {
+	return newServicerAndStorager(pairs...)
+}
+
+func NewServicer(pairs ...types.Pair) (types.Servicer, error) {
+	return newServicer(pairs...)
 }
 
 // NewStorager will create Storager only.
 func NewStorager(pairs ...types.Pair) (types.Storager, error) {
-	panic("implement me")
+	_, store, err := newServicerAndStorager(pairs...)
+	return store, err
+}
+
+func newServicer(pairs ...types.Pair) (srv *Service, err error) {
+	defer func() {
+		if err != nil {
+			err = services.InitError{Op: "new_servicer", Type: Type, Err: formatError(err), Pairs: pairs}
+		}
+	}()
+
+	srv = &Service{}
+
+	opt, err := parsePairServiceNew(pairs)
+	if err != nil {
+		return nil, err
+	}
+
+	cp, err := credential.Parse(opt.Credential)
+	if err != nil {
+		return nil, err
+	}
+	if cp.Protocol() != credential.ProtocolHmac {
+		return nil, services.PairUnsupportedError{Pair: ps.WithCredential(opt.Credential)}
+	}
+	ak, sk := cp.Hmac()
+
+	ep, err := endpoint.Parse(opt.Endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	var url string
+	var secure bool
+	switch ep.Protocol() {
+	case endpoint.ProtocolHTTP:
+		url, _, _ = ep.HTTP()
+		url = url[7:]
```

Comment:
What does this for? How about constructing them via `host:port`?

Reply:
`minio.New` needs the endpoint without protocal label. The field `secure` handles this as an option.

Reply:
`url[7:]` looks magic number to me. How about using the host and port that returned by `ep.HTTP()` to construct the addr?

Or leave a comment to make it clear.

Reply:
`http://`and `https://`

Reply:
> `http://`and `https://`

Well, as your reviewer, I do understand that does `url[7:]` means. But for other developers, it lacks readability.

Add clear comments here or rewrite into the following style both looks good to me :

```go
_, host, port := ep.HTTP()
addr := fmt.Sprintf("%s:%s", host, port)
```